keras_nlp/tokenizers/word_piece_tokenizer.py

```diff
@@ -149,11 +164,33 @@ def pretokenize(
         text = tf.strings.regex_replace(text, r"\p{Mn}", "")
     if split:
         if split_on_cjk:
-            split_pattern = WHITESPACE_PUNCTUATION_AND_CJK_REGEX
-            keep_split_pattern = PUNCTUATION_AND_CJK_REGEX
+            if special_tokens_pattern is not None:
```

Comment:
seems like you could shorten this logic a bit...
```python
if split_on_cjk:
    split_pattern = WHITESPACE_PUNCTUATION_AND_CJK_REGEX
    keep_split_pattern = PUNCTUATION_AND_CJK_REGEX
else:
    split_pattern = WHITESPACE_AND_PUNCTUATION_REGEX
    keep_split_pattern = PUNCTUATION_REGEX
if special_tokens_pattern is not None:
     split_pattern = r"|".join(special_tokens_pattern, split_pattern)
     keep_split_pattern = r"|".join(special_tokens_pattern, keep_split_pattern)
```
something like that

Reply:
Thanks!